components/bookshelf/discover/SearchBox.tsx

```diff
@@ -16,19 +23,26 @@ interface SearchBoxProps {
 const SearchBox = ({ setIsSearching }: SearchBoxProps) => {
   const [filteredData, setFilteredData] = useState()
```

Comment:
```suggestion
  const [filteredData, setFilteredData] = <BookType[]>useState([])
```

Reply:
ohh 😮this one fixed it. great catch! can't believe I missed that one